mediapipe/graphs/instantmotiontracking/calculators/gl_animation_overlay_calculator.cc

```diff
@@ -402,6 +309,67 @@ bool GlAnimationOverlayCalculator::LoadAnimationAndroid(
       LOG(ERROR) << "Failed to read indices for frame " << frame_count_;
       return false;
     }
+
+    // Set triangle_mesh normals for shader usage
+    int len = lengths[2];
+    triangle_mesh.normals.reset(new float[len]);
+    float triangle_normals[len];
+
+    // Compute every triangle surface normal and store them for later use.
+    for (int idx = 0; idx < len; idx += 3) {
+      int v1 = triangle_mesh.triangle_indices.get()[idx];
+      int v2 = triangle_mesh.triangle_indices.get()[idx + 1];
+      int v3 = triangle_mesh.triangle_indices.get()[idx + 2];
+      // Previous (V1) vertex X,Y,Z indices in triangle_mesh.vertices
+      float V1x = triangle_mesh.vertices[v1 * 3];
+      float V1y = triangle_mesh.vertices[v1 * 3 + 1];
+      float V1z = triangle_mesh.vertices[v1 * 3 + 2];
+      // Current (V2) vertex X,Y,Z indices in triangle_mesh.vertices
+      float V2x = triangle_mesh.vertices[v2 * 3];
+      float V2y = triangle_mesh.vertices[v2 * 3 + 1];
+      float V2z = triangle_mesh.vertices[v2 * 3 + 2];
+      // Next (V3) vertex X,Y,Z indices in triangle_mesh.vertices
+      float V3x = triangle_mesh.vertices[v3 * 3];
+      float V3y = triangle_mesh.vertices[v3 * 3 + 1];
+      float V3z = triangle_mesh.vertices[v3 * 3 + 2];
+      // Calculate normals from vertices
+      // V2 - V1
+      float Ax = V2x - V1x;
+      float Ay = V2y - V1y;
+      float Az = V2z - V1z;
+      // V3 - V1
+      float Bx = V3x - V1x;
+      float By = V3y - V1y;
+      float Bz = V3z - V1z;
+      // Calculate cross product
+      triangle_normals[idx] = Ay * Bz - Az * By;
+      triangle_normals[idx + 1] = Az * Bx - Ax * Bz;
+      triangle_normals[idx + 2] = Ax * By - Ay * Bx;
+    }
+
+    // Normalize each triangle surface normal
+    for(int i = 0; i < len; i+=3) {
+      float product = 0.0;
+      product = product + triangle_normals[i] * triangle_normals[i];
+      product = product + triangle_normals[i+1] * triangle_normals[i+1];
+      product = product + triangle_normals[i+2] * triangle_normals[i+2];
+      float magnitude = sqrt(product);
+      triangle_normals[i] /= magnitude;
```

Comment:
[No action necessary] We can assume our assets are well-formed because we're creating them.  But in general, this would be a good thing to have a debug-mode-only check for (specifically that magnitude isn't 0 so we don't divide by 0).